### ISSUE TITLE
`aio login` without a browser (paste token)

### ISSUE THREAD

**Issue by shazron** [adobe/aio-cli-plugin-auth#54]
See how the Github CLI does this: https://github.com/cli/cli/pull/1445
Not sure if possible with Adobe IMS.

**Comment by shazron** [comment]
yeah currently we would make them paste the token in the `.env` at a certain key, this way they can just paste the token via the cli (so we skip one possibly error-prone step)

**Comment by meryllblanchet** [comment]
I like the idea, wdyt @sarahxxu ?

**Comment by fmeschbe** [comment]
There is catches here:

- Our tokens all have a limited expiry time of ~24h (depending on Client ID configuration)
- We do not have the GitHub feature of creating "personal" access tokens which are valid until deleted.

So, I fear, this is not going to work for now. Maybe in the future, but currently, the only way to get a personal access token is using the 3-legged HTML based OAuth flow.

BTW: Anyone tried Lynx ? (does it still exist after all ?)

**Comment by meryllblanchet** [comment]
@fmeschbe and I discussed offline, and the idea to provide an option like `aio login --store` for CLI users who do not have an org access and get a tech account token from an org admin/developer, would still be applicable.

This CLI user persona (most likely a contractor not belonging to an IMS org), could then get their temporary tech account token persisted at the right place in the CLI configuration.

They would however not get logged in via IMS using that feature, but could reuse the token for CLI operations that would require it.

@fmeschbe the latest Lynx release is from 2018, and looks like they're working on a [2.9.0](https://lynx.invisible-island.net/current/CHANGES.html) :-)

**Comment by shazron** [comment]
The use case is precisely what Meryll described. As for token expiry, `aio login` also gives  a refresh token, and that can be passed on as well. In that case, the input wouldn't just be a token, but perhaps a base64 encoded object that contains both access and refresh tokens (that was previously exported by running `aio config:get ims.contexts.cli` for example

**Comment by shazron** [comment]
As for text mode browsers, try https://brow.sh ;)

**Comment by hbitdeveloper** [comment]
Hello guys, Have you found a solution yet? I paste the token in the .env at a certain key, But still unable to work.

**Comment by shazron** [comment]
@hbitdeveloper what key did you use?

**Comment by hbitdeveloper** [comment]
I have already logged in to the Windows system,Then an .env file will be generated in my folder.
So I copied this file to the Linux server.Then execute the 'aio login' command. But it still doesn't work

**Comment by shazron** [comment]
ah ok, no you need to "paste the token in the .env at a certain key" not "copy the .env". I'll have to get back to you regarding which key (this key does not exist in the .env)

**Comment by shazron** [comment]
The alternative in a pinch:

Login on the desktop (somewhere), then export the json value from the project:
`aio config:get ims.contexts.cli > token.json`

Then import that .json into your project:
`aio config:set --json ims.contexts.cli < token.json`
